### PR TITLE
Add mount retract RC switch option in Plane

### PR DESCRIPTION
**Pull request by Hwurzburg**
somehow this was missed at some point...if you have a servo mount, you may need this, if for nothing else but to stop stabilization

### CODE REVIEW

File: ArduPlane/RC_Channel.cpp [hunk 1]
```diff
@@ -281,6 +282,22 @@ void RC_Channel_Plane::do_aux_function(const aux_func_t ch_option, const AuxSwit
        do_aux_function_flare(ch_flag);
        break;
 
+   case AUX_FUNC::RETRACT_MOUNT:
```
Comment (peterbarker) [inline ArduPlane/RC_Channel.cpp:285]: Shouldn't this be going up into the common library?

Should be able to use `AP::mount()` to get the mount.

Comment (Hwurzburg) [reply ArduPlane/RC_Channel.cpp:285]: I just copied what Copter does...not sure what you want me to do here...

Comment (peterbarker) [reply ArduPlane/RC_Channel.cpp:285]: > I just copied what Copter does...not sure what you want me to do here...

Generally we really, really don't like copying code.  It means you end up with twice the number of bugs.

What we're after is moving the code from ArduCopter's RC_Channel library into the common RC_Channel library.

You actually helped doing this once before:

```
commit 4a51de724d228ac06f503b8cb875b52f5c8d112c
Author: Henry Wurzburg <hwurzburg@yahoo.com>
Date:   Sat Nov 2 23:18:34 2019 -0500

    Copter: move support for ADSB Avoidance enable up to RC_Channel

commit 1238d4b3e6dd6ef3e596058366dba5fd0d736198
Author: Henry Wurzburg <hwurzburg@yahoo.com>
Date:   Sat Nov 2 23:14:39 2019 -0500

    RC_Channel: add ADSB Avoidance enable option to RC options for plane

commit e12a841eb579f20331144377195ea1a60fecbe17
Author: Peter Barker <pbarker@barker.dropbear.id.au>
Date:   Mon Nov 4 11:18:01 2019 +1100

    RC_Channel: move support for ADSB avoidance switch enable up
```

Comment (Hwurzburg) [reply ArduPlane/RC_Channel.cpp:285]: oh that common...I thought you meant some AP_Common....np...can move it into RC_Channel...

Comment (Hwurzburg) [reply ArduPlane/RC_Channel.cpp:285]: done

File: ArduCopter/RC_Channel.cpp [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 #include "RC_Channel.h"
 
+
```
Comment (peterbarker) [inline ArduCopter/RC_Channel.cpp:5]: Stray whitespace addition

File: ArduCopter/RC_Channel.cpp [hunk 3]
```diff
@@ -118,6 +118,7 @@ void RC_Channel_Copter::init_aux_function(const aux_func_t ch_option, const AuxS
     case AUX_FUNC::AIRMODE:
         do_aux_function_change_air_mode(ch_flag);
         break;
+
```
Comment (peterbarker) [inline ArduCopter/RC_Channel.cpp:121]: Stray whitespace addition

File: libraries/RC_Channel/RC_Channel.cpp [hunk 4]
```diff
@@ -95,7 +96,8 @@ const AP_Param::GroupInfo RC_Channel::var_info[] = {
...
+    // @Values{Plane}: 0:Do Nothing, 4:ModeRTL, 9:Camera Trigger, 16:ModeAuto, 24:Auto Mission Reset, 27:Retract Mount, 28:Relay On/Off, 29:Landing Gear, 30:Lost Plane Sound, 31:Motor Emergency Stop, 34:Relay2 On/Off, 35:Relay3 On/Off, 36:Relay4 On/Off, 38:ADSB Avoidance En, 41:ArmDisarm, 43:InvertedFlight, 46:RC Override Enable, 51:ModeManual, 55:ModeGuided, 56:ModeLoiter, 58:Clear Waypoints, 62:Compass Learn, 64:Reverse Throttle, 65:GPS Disable, 66:Relay5, 67:Relay6, 72:ModeCircle, 77:ModeTakeoff, 78:RunCam Control, 79:RunCam OSD Control, 81:Disarm, 82:QAssist 3pos, 84:Air Mode, 85:Generator, 86: Non Auto Terrain Follow Disable, 87:Crow Select, 88:Soaring Enable, 89:Landing Flare, 90:EKF Pos Source, 100:KillIMU1, 101:KillIMU2, 102:Camera Mode Toggle, 105:GPS Disable Yaw, 208:Flap, 209: Forward Throttle, 300:Scripting1, 301:Scripting2, 302:Scripting3, 303:Scripting4, 304:Scripting5, 305:Scripting6, 306:Scripting7, 307:Scripting8
```
Comment (peterbarker) [inline libraries/RC_Channel/RC_Channel.cpp:99]: Rover and Sub will also get this feature.

Sub doesn't do RC options yet - but Rover should ge tthe entry in its {} here.

File: libraries/RC_Channel/RC_Channel.cpp [hunk 5]
```diff
@@ -1074,6 +1079,27 @@ void RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos
         break;
     }
 
+    case AUX_FUNC::RETRACT_MOUNT: {
+#if HAL_MOUNT_ENABLED
```
Comment (peterbarker) [inline libraries/RC_Channel/RC_Channel.cpp:1083]: At some stage we need to canonicalise on whether the `#if` goes on the outside of the case statement.

If you don't put it on the inside like this then any time somebody tries the switch they'll get a message.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Should be just `Copter:` in the commit message.

Looks the right shape

**Comment by Hwurzburg** [comment]
@peterbarker Someone needs to fix the git-subsystems-split tool to fix ArduCopter vs Copter

File: libraries/RC_Channel/RC_Channel.cpp (hunk 5 above)
Comment (Hwurzburg) [reply libraries/RC_Channel/RC_Channel.cpp:1083]: I followed the copter pattern...do you want me to change it? if the mount is not built, wouldn't you want to have the failed init message if someone setup the switch? so the if define should be outside, right?

File: libraries/RC_Channel/RC_Channel.cpp (hunk 4 above)
Comment (Hwurzburg) [reply libraries/RC_Channel/RC_Channel.cpp:99]: added

File: ArduCopter/RC_Channel.cpp (hunk 3 above)
Comment (Hwurzburg) [reply ArduCopter/RC_Channel.cpp]: fixed

File: ArduCopter/RC_Channel.cpp (hunk 2 above)
Comment (Hwurzburg) [reply ArduCopter/RC_Channel.cpp:5]: fixed

**Comment by peterbarker** [comment]
On Sun, 22 Nov 2020, Henry Wurzburg wrote:

> I followed the copter pattern...do you want me to change it? if the mount is
> not built, wouldn't you want to have the failed init message if someone
> setup the switch? so the if define should be outside, right?

I'd think so.  But I'm not sure what our policy is, so what you've got is 
fine.  If you'd like to get the message out, change it :)

**Review by peterbarker — APPROVED** [review]

**Review by rmackay9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**